test/micro/org/openjdk/bench/jdk/classfile/ConstantPoolBuildingClassEntry.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.bench.jdk.classfile;
+
+import java.lang.classfile.constantpool.ConstantPoolBuilder;
+import java.lang.constant.ClassDesc;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+import jdk.internal.classfile.impl.Util;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+
+import static java.lang.constant.ConstantDescs.*;
+
+import static org.openjdk.bench.jdk.classfile.TestConstants.*;
+
+/**
+ * Tests constant pool builder lookup performance for existing entries.
+ */
+@Warmup(iterations = 3)
+@Measurement(iterations = 5)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@BenchmarkMode(Mode.Throughput)
+@Fork(value = 1, jvmArgsAppend = {
+        "--enable-preview",
+        "--add-exports", "java.base/jdk.internal.classfile.impl=ALL-UNNAMED"})
+@State(Scope.Benchmark)
+public class ConstantPoolBuildingClassEntry {
+    // JDK-8338546
+    static final int RUNS = 0xff; // Q: Should we measure consecutive runs at all?
+    ConstantPoolBuilder builder;
+    List<ClassDesc> classDescs;
+    List<ClassDesc> nonIdenticalClassDescs;
+    List<String> internalNames;
+    Random random;
+    int size;
+
+    @Setup(Level.Iteration)
+    public void setup() {
+        builder = ConstantPoolBuilder.of();
+        classDescs = List.of(
+                CD_Byte, CD_Object, CD_Long.arrayType(), CD_String, CD_String, CD_Object, CD_Short,
```

Comment:
Should this include some primitive class descs? I only see wrapper types like `CD_Integer`

Reply:
Unfortunately primitive types aren't encoded by class entries, but accessed by getstatic on Wrapper's TYPE field.

Reply:
Ok, good. Maybe add a comment that this collection of descs deliberately omits primitive types since they would not usually end up in this code?

Reply:
Uh, I mean it's a class entry... and class file format prohibits primitives in class entries. I would deduce that information from the benchmark name.

Reply:
Yes, obvious in hind-sight, but comments are good to remind readers of context that might not be obvious at a glance or without paging in the full context.

Reply:
Added a note in this benchmark; should suffice.